osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SoapEndpointDataTypeConverter.java

```diff
@@ -16,100 +16,93 @@
         return org.opensmartgridplatform.ws.schema.core.secret.management.SecretType.fromValue(secretType.name());
     }
 
     private org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret encryptAndConvertSoapTypedSecret(
-            TypedSecret typedSecret) {
-        org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret soapTypedSecret =
+            final TypedSecret typedSecret) {
+        final org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret soapTypedSecret =
                 new org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret();
 
-        String encodedSecret = typedSecret.getSecret();
-        byte[] rawSecret = HexUtils.fromHexString(encodedSecret);
-        Secret secret = new Secret(rawSecret);
-        EncryptedSecret encryptedSecret = encryptionDelegate.encrypt(EncryptionProviderType.RSA, secret, KEY_REFERENCE);
-        soapTypedSecret.setSecret(HexUtils.toHexString(encryptedSecret.getSecret()));
+        //final String encodedSecret = typedSecret.getSecret();
+        final byte[] rsaSecret = typedSecret.getSecret(); //HexUtils.fromHexString(encodedSecret);
+        //final EncryptedSecret encryptedSecret = this.encryptionDelegate.encrypt(EncryptionProviderType.RSA,
+        //        rawSecret, KEY_REFERENCE);
+        soapTypedSecret.setSecret(HexUtils.toHexString(rsaSecret));
 
-        SecretType secretType = typedSecret.getSecretType();
-        soapTypedSecret.setType(convertToSoapSecretType(secretType));
+        final SecretType secretType = typedSecret.getSecretType();
+        soapTypedSecret.setType(this.convertToSoapSecretType(secretType));
 
         return soapTypedSecret;
     }
 
-    private TypedSecret decryptAndConvertSoapTypedSecret(
-            org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret soapTypedSecret) {
-        TypedSecret typedSecret = new TypedSecret();
+    public TypedSecret decryptAndConvertSoapTypedSecret(
+            final org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecret soapTypedSecret) {
 
-        byte[] rawEncryptedSecret = HexUtils.fromHexString(soapTypedSecret.getSecret());
-        EncryptedSecret encryptedSecret = new EncryptedSecret(EncryptionProviderType.RSA, rawEncryptedSecret);
-        Secret decryptedSecret = encryptionDelegate.decrypt(encryptedSecret, KEY_REFERENCE);
+        final byte[] rsaEncryptedSecret = HexUtils.fromHexString(soapTypedSecret.getSecret());
+        //final EncryptedSecret encryptedSecret = new EncryptedSecret(EncryptionProviderType.RSA, rawEncryptedSecret);
+        //final byte[] decryptedSecret = this.encryptionDelegate.decrypt(encryptedSecret, KEY_REFERENCE);
 
-        typedSecret.setSecret(HexUtils.toHexString(decryptedSecret.getSecret()));
-        typedSecret.setSecretType(convertToSecretType(soapTypedSecret.getType()));
+        final TypedSecret typedSecret = new TypedSecret(rsaEncryptedSecret,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "typedSecret". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1488)